### PR TITLE
Move to Github Actions

### PR DESCRIPTION
**Pull request by dwilkie**
I noticed that the travis build is no longer working so I thought it might be a good opportunity to switch to Github Actions.

If you don't want to switch feel free to just close this PR. Otherwise leave a comment and I'll continue working on it.

https://github.com/somleng/adhearsion/runs/5282043628?check_suite_focus=true

Todo:

* [x] Configure (or remove) code climate (No changes needed)
* [x] Fix cucumber
* [x] Remove .travis.yml
* [x] Test against latest Ruby 2.x and 3.x (depends on https://github.com/adhearsion/adhearsion/pull/654)

### CODE REVIEW

**Comment by bklang** [comment]
After a bit of research into this, it appears that Travis isn't a good fit for Adhearsion right now. Thanks for making a start on the migration to Github Actions

**Comment by dwilkie** [comment]
I think this is done. Github Actions is configured to run against the latest ruby `2.x` and `3.x` versions. Here's a link to the build on my fork. https://github.com/somleng/adhearsion/runs/5282043628?check_suite_focus=true

File: .github/workflows/build.yml
Comment (benlangfeld) [inline .github/workflows/build.yml]: For this to work correctly on this repo, where most PRs come from forks, you need to do `on: pull_request`.

```suggestion
on:
  push:
    branches:
      - master
  pull_request:
```

File: .github/workflows/build.yml
Comment (benlangfeld) [inline .github/workflows/build.yml]: Can we not be more specific about versions here like we were being on Travis?

Comment (benlangfeld) [inline .github/workflows/build.yml]: You also removed testing on JRuby, which is important to this project.

**Review by benlangfeld — CHANGES_REQUESTED** [review]
Just a few things I'd prefer to polish up before merging this.

File: .github/workflows/build.yml
Comment (dwilkie) [reply .github/workflows/build.yml]: Sure. I just thought it might be better to say that we want to always test against the latest ruby version. If not, the matrix will grow, but it's not always beneficial. For example why do we need to test against ruby `2.7.3` and ruby `2.7.4`. Maybe we could just test agains the latest in the minor version field, e.g. `2.7.x` and `3.1.x`?

Comment (dwilkie) [reply .github/workflows/build.yml]: I added `jruby` and `jruby-head` but the build is failing for these due to an error when running `bundle install`

Comment (benlangfeld) [reply .github/workflows/build.yml]: > Maybe we could just test agains the latest in the minor version field, e.g. `2.7.x` and `3.1.x`?

Yes, that's fine, but we do typically want to test multiple of those minor versions.

File: .github/workflows/build.yml
Comment (benlangfeld) [inline .github/workflows/build.yml:3]: Any idea why this didn't run here?

Comment (bklang) [reply .github/workflows/build.yml:3]: It looks like the actions are executing on the [forked repo](https://github.com/somleng/adhearsion/actions/runs/1879253934). Strange that they do not appear here as well.

File: .github/workflows/build.yml
Comment (bklang) [reply .github/workflows/build.yml]: This is a known issue. The last version of  the `http_parser.rb` gem that was released for JRuby was 0.6.0. There are known issues in later versions, the most recent work is appearing [on this PR](https://github.com/tmm1/http_parser.rb/pull/75). The fix is to lock `http_parser.rb` to `~> 0.6.0`

Comment (bklang) [reply .github/workflows/build.yml]: Related: after the bundle install works, you very likely will see failing test runs under JRuby because of [another bug](https://github.com/TwP/logging/issues/235) I just documented against the Logging gem. The workaround for that is to pin `logging` gem to `~> 2.2.2`

File: .github/workflows/build.yml
Comment (bklang) [inline .github/workflows/build.yml:15]:
```suggestion
        ruby: ['2.7', '3.1', 'jruby-9.1', 'jruby-9.2', 'jruby-9.3', 'jruby-head']
```

Suggest this because JRuby 9.1.17.0 is known to pass tests, while 9.2 and later have breakage. We should allow 9.2/9.3/jruby-head to fail for now, until we can fix those tests.

Comment (bklang) [reply .github/workflows/build.yml:15]: Oops, I've been slandering JRuby unfairly. It turns out the specs fail with `nokogiri` starting at `1.11`. If we lock to `~> 1.8, >= 1.8.3, < 1.11` then the specs pass with all currently released versions of JRuby.

I think there are still some problems with the Cucumber console tests with newer JRuby, will look into that later.

File: .github/workflows/build.yml
Comment (benlangfeld) [reply .github/workflows/build.yml:3]: That's a prerequisite to merging this.

File: .github/workflows/build.yml
Comment (bklang) [reply .github/workflows/build.yml]: I've added these constraints under a separate PR here: https://github.com/adhearsion/adhearsion/pull/656/files.  When that merges we can include it on this branch.

File: .github/workflows/build.yml
Comment (dwilkie) [reply .github/workflows/build.yml:3]: I think this is not running because the upstream repo (this one) is not yet setup for Github actions yet. Once this is merged, future pull requests should run.

Comment (benlangfeld) [reply .github/workflows/build.yml:3]: This repo has the correct configuration. I don't think these manifests need to be on the main branch for it to run.

Comment (benlangfeld) [reply .github/workflows/build.yml:3]: Screenshot 2022-03-01 at 09 51 37

Comment (bklang) [reply .github/workflows/build.yml:3]: https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks
This seems relevant, though I can't immediately see why we don't have the button to approve the workflow on this PR.

Comment (benlangfeld) [reply .github/workflows/build.yml:3]: Because @dwilkie is not a first time contributor. It should run without approval.

Comment (benlangfeld) [reply .github/workflows/build.yml:3]: Not even setting "Require approval for all outside collaborators" makes the button show up, though.

Comment (bklang) [reply .github/workflows/build.yml:3]: I can't see any reason these actions aren't running. Since Travis is already broken, any objection to merging this PR as-is and see if that's what kickstarts Github into running the actions?

Comment (benlangfeld) [reply .github/workflows/build.yml:3]: If we're all out of ideas, I guess that's all we can do :)

**Review by benlangfeld — APPROVED** [review]

**Pull request merged**
